src/reconcile/rec_visibility.c

```diff
@@ -252,6 +257,81 @@ __timestamp_out_of_order_fix(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw
     return (false);
 }
 
+/*
+ * __rec_validate_upd_chain --
+ *     Check the update chain for conditions that would prevent it's insertion into the history
+ *     store. Return EBUSY if the update chain cannot be inserted into the history store at this
+ *     time.
+ */
+static int
+__rec_validate_upd_chain(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *select_upd,
```

Comment:
I'm not sure if fixed length column store will go to this path or not. Different from other cases, we don't free anything on the update chain until they are globally visible. I think we should bypass this code for fixed length column store as it doesn't have history store enabled. Similar for the in memory database case.

Reply:
@keithbostic I think this one is for you, since I know you've been thinking about FLCS in a timestamp world recently.

Reply:
(Oops, didn't see this until now. Yeah, there should be no need for special-case code going forward; I've written a good chunk of the code, and while it doesn't all work yet I haven't hit any major stoppers.)